modules/ztp-image-based-upgrade-prep.adoc

```diff
@@ -0,0 +1,105 @@
+// Module included in the following assemblies:
+// Epic TELCOSTRAT-160 (4.15/4.16), story TELCODOCS-1576
+// * scalability_and_performance/ztp-talm-updating-managed-policies.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ztp-image-based-upgrade-prep_{context}"]
+= Preparing the {sno} clusters for the image-based upgrade
+
+When you deploy the {lcao} on a cluster, a blank `imageBasedUpgrade` CR is automatically created.
+You use this CR to specify the image repository of the seed image and to move through the different stages.
+
+In the `Prep` stage, the {lcao} ensures that the target clusters are ready for the upgrade by checking if they meet certain conditions. The {lcao} also precaches any additional listed images on the target clusters from the seed image.
+You can complete this stage well before starting the upgrade.
+
+After you start the `Prep` stage by applying the `imageBasedUpgrade` CR with {cgu-operator}, {cgu-operator} applies a policy that starts the following procedures:
```

Comment:
Not sure how deep we want to go on applying things with TALM. Written like this for a user that does not know anything about TALM/ZTP/GitOps it cannot be understood. Looks like TALM applies CRs. Here it is stated that moving to Prep requires TALM however in the example below it shows a patch command to modify the Stage spec from Idle to Prep in the target cluster. That is confusing.


I would try to briefly detail/add:

* A link to TALM and GitOps ZTP process
* Explain that it can applied in the target cluster manually as shown in the patch command, but if you're running a fleet of clusters that want to upgrade a GitOps approach is more convenient. In that case, when your target clusters are managed by a ACM hub cluster, you can use GitOps ZTP to create policies that can be enforced and targeted to the clusters you decide via TALM. In that case, the dataprotectionApplication and the secret configuration can be applied to one or multiple clusters in a batch of 1 or more.  

To give more context, this GitOps approach + TALM can be done with almost every change you want to apply: moving IBU through the different stages, install OADP, apply OADP backups & restore config, install LCA... it makes sense when you're dealing with a fleet of clusters and you want to manage the software/configuration lifecycle of your managed clusters. Hope it helps instead of being more confusing.

Reply:
It helps, thank you for writing out the explanation! I will remove this for now and come back to connecting it to TALM/ZTP, once the workflow itself is cleaned up.

Reply:
@alosadagrande If the user is doing it without TALM, "who" creates the DPA CR and the secret to the bucket? Is that all manually done by the user or (partially) automated by OADP?

Reply:
It is not automated by OADP.  It is part of any OADP configuration, so it must be applied either by the user or it can be applied using GitOps (e.g. Argo) or any automation tool that the user has.